[stable/traefik] Decouple prometheus metrics to dashboard

mazzy89 opened the pull request:
<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:

Before this PR, in case the dashboard was not enabled, the Prometheus metrics were disabled.

It is now possible to enable Prometheus metrics having at the same time the dashboard disabled.

I've fully tested the PR locally and it works as expected.

Signed-off-by: Salvatore Mazzarino <dev@mazzarino.cz>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

szymonpk commented on stable/traefik/templates/service.yaml:
```diff
@@ -55,7 +55,7 @@ spec:
     targetPort: httpn
     {{- end }}
   {{- if (and (.Values.metrics.prometheus.enabled) (not (.Values.metrics.prometheus.restrictAccess)))}}
-  - port: 8080
```
I am not person eligible for the review, but I wanted to throw my 3 cents. Shouldn't this be a completely separate service? I would like to keep metrics internal for the cluster.

mazzy89 replied:
I vote for this. I can separate the `Service`

chelnak replied:
Yes agree,. I don't want my metrics accessible publicly.

mazzy89 replied:
I'll do this change tonight

mazzy89 replied:
I've double-checked this and it is fine. In case that Prometheus is enabled `Service` has 9100 port but this is not exposed anyhow to the external. Indeed the dashboard `Ingress` points to a different `Service` so @szymonpk to reply to you I think it is not strictly needed a separate `Service` for this.

My concern here is of a different nature. This PR should be address asap because it is a pretty serious security risk. I can't imagine how many Traefik dashboards are exposed out there for all who have Prometheus metrics enabled.

I hope maintainers are aware of this.

szymonpk replied:
Aren't all ports exported to the external ELB? I believe they were, when I tried to use it last time.

mazzy89 replied:
I've doubled check this and actually all the ports are exposed to the ELB. I'll make another round of changes and wire a PR for today

mazzy89 committed:
[stable/traefik] Decouple prometheus metrics to dashboard

Before this PR, in case the dashboard was not enabled, the Prometheus metrics were disabled.

It is now possible to enable Prometheus metrics having at the same time the dashboard disabled.

Signed-off-by: Salvatore Mazzarino <dev@mazzarino.cz>

vdemeester reviewed (commented):
/lgtm
